notebooks/115-async-api/115-async-api.ipynb

```diff
@@ -0,0 +1,460 @@
+    "            scale = 1280 / max(frame.shape)\n",
+    "            if scale < 1:\n",
+    "                frame = cv2.resize(frame, None, fx=scale, fy=scale, interpolation=cv2.INTER_AREA)\n",
+    "            resized_frame = preprocess(frame)\n",
+    "            infer_request.set_tensor(input_layer_ir, ov.Tensor(resized_frame))\n",
+    "            # Start the inference request in sync\n",
+    "            infer_request.infer()\n",
+    "            res = infer_request.get_output_tensor(0).data\n",
+    "            stop_time = time.time()\n",
+    "            total = stop_time - start_time\n",
+    "            num = num + 1\n",
+    "            sync_fps = num / total \n",
+    "            frame = postprocess(res, frame, sync_fps)\n",
+    "            # Display the results\n",
+    "            if use_popup:\n",
+    "                cv2.imshow(title, frame)\n",
+    "                key = cv2.waitKey(1)\n",
+    "                # escape = 27\n",
+    "                if key == 27:\n",
+    "                    break\n",
+    "            else:\n",
+    "                # Encode numpy array to jpg\n",
+    "                _, encoded_img = cv2.imencode(\".jpg\", frame, params=[cv2.IMWRITE_JPEG_QUALITY, 90])\n",
+    "                # Create IPython image\n",
+    "                i = display.Image(data=encoded_img)\n",
+    "                # Display the image in this notebook\n",
+    "                display.clear_output(wait=True)\n",
+    "                display.display(i)         \n",
+    "    # ctrl-c\n",
+    "    except KeyboardInterrupt:\n",
+    "        print(\"Interrupted\")\n",
+    "    # Any different error\n",
+    "    except RuntimeError as e:\n",
+    "        print(e)\n",
+    "    finally:\n",
+    "        if player is not None:\n",
+    "            # stop capturing\n",
+    "            player.stop()\n",
+    "            return sync_fps\n",
+    "        if use_popup:\n",
+    "            cv2.destroyAllWindows()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Test the perfromance in Sync Mode"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sync_fps = sync_api(source=video_path, flip=False, fps=30, use_popup=False, skip_first_frames=1000)\n",
+    "print(f\"average throuput in sync mode: {sync_fps:.2f} fps\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Async Mode (async-based approach)\n",
+    "\n",
+    "Let’s see how the OpenVINO Async API can improve overall frame rate of the application. The key advantage of the Async approach is as follows: while a device is busy with the inference, the application can do other things in parallel (e.g. populating inputs or scheduling other requests) rather than wait for the current inference to complete first.\n",
```

Comment:
```suggestion
    "Let’s see how the OpenVINO Async API can improve the overall frame rate of an application. The key advantage of the Async approach is as follows: while a device is busy with the inference, the application can do other things in parallel (e.g. populating inputs or scheduling other requests) rather than wait for the current inference to complete first.\n",
```